src/TreeBuilder.cpp

```diff
@@ -215,8 +214,9 @@ void TreeBuilder::build(const std::vector<uint64_t>& data,
     auto& rootID = rootIDs.emplace_back(nextNodeID++);
 
     try {
-      // The first value is the address of the root object
-      pointers.insert(next());
+      // The first value is the address of the root object. Drop it as we don't
+      // need to manage pointer deduplication anymore.
+      next();
```

Comment:
Can we just stop storing this pointer in the first place?

Reply:
Yes. I was a little worried as I'm not sure I've got every entrypoint where it's written, but I put it in a separate commit so it can be easily reverted.